packages/kbn-flot-charts/lib/jquery_flot.js

```diff
@@ -1,3168 +1,3777 @@
+          pos.left = m[0] + plotOffset.left + 'px';
+        }
+        const legend = $('<div></div>')
+          .addClass('legend')
+          .append(table.css(pos))
+          .appendTo(placeholder);
+        if (options.legend.backgroundOpacity !== 0.0) {
+          // put in the transparent background
+          // separately to avoid blended labels and
+          // label boxes
+          let c = options.legend.backgroundColor;
+          if (c == null) {
+            c = options.grid.backgroundColor;
+            if (c && typeof c === 'string') {
+              c = $.color.parse(c);
             } else {
-                placeholder.find(".legend").remove();
-            }
-
-            if (!options.legend.show) {
-                return;
-            }
-
-            var fragments = [], entries = [], rowStarted = false,
-                lf = options.legend.labelFormatter, s, label;
-
-            // Build a list of legend entries, with each having a label and a color
-
-            for (var i = 0; i < series.length; ++i) {
-                s = series[i];
-                if (s.label) {
-                    label = lf ? lf(s.label, s) : s.label;
-                    if (label) {
-                        entries.push({
-                            label: label,
-                            color: s.color
-                        });
-                    }
-                }
-            }
-
-            // Sort the legend using either the default or a custom comparator
-
-            if (options.legend.sorted) {
-                if ($.isFunction(options.legend.sorted)) {
-                    entries.sort(options.legend.sorted);
-                } else if (options.legend.sorted == "reverse") {
-                	entries.reverse();
-                } else {
-                    var ascending = options.legend.sorted != "descending";
-                    entries.sort(function(a, b) {
-                        return a.label == b.label ? 0 : (
-                            ((a.label < b.label) != ascending ? 1 : -1)   // Logical XOR
-                        );
-                    });
-                }
-            }
-
-            // Generate markup for the list of entries, in their final order
-
-            for (var i = 0; i < entries.length; ++i) {
-
-                var entry = entries[i];
-
-                if (i % options.legend.noColumns == 0) {
-                    if (rowStarted)
-                        fragments.push('</tr>');
-                    fragments.push('<tr>');
-                    rowStarted = true;
-                }
-
-                fragments.push(
-                    '<td class="legendColorBox"><div style="border:1px solid ' + options.legend.labelBoxBorderColor + ';padding:1px"><div style="width:4px;height:0;border:5px solid ' + entry.color + ';overflow:hidden"></div></div></td>' +
-                    '<td class="legendLabel" data-test-subj="flotLegendLabel">' + entry.label + '</td>'
```

Comment:
This is the fix I really wanted to see. No more constructing divs from string templates! 💯